### ISSUE TITLE
App not working on macOS Monterey; Says Insufficient Permissions, provide Input Monitoring permissions

### ISSUE THREAD

**Issue by pratyushmanocha** [rnayabed/rangoli#24]
Hi, 

I am using a 15" MacBook Pro with Touch Bar and TouchID from 2016 running macOS Monterey. I have connected my Royal Kludge RK100 RGB with BT and 2.4G Wireless keyboard using the wire that came with it. The Rangoli app is recognising the keyboard properly but is unable to change its settings. 

Every time I start the app, it prompts me to provide it Input Monitoring permissions through my laptop settings (System Preferences > Security & Privacy > Privacy > Input Monitoring). I have already done that and have tried restarting both the application and my laptop, yet this error does not go away. 

I started the app and tried to change the RGB effect. The corresponding log file is attached below... Any idea how to rectify it? 

Thanks in advance!

[rangoli.log](https://github.com/rnayabed/rangoli/files/11130776/rangoli.log)

**Comment by pratyushmanocha** [comment]
Also relevant: rangoli doesn't automatically show up in the Input Monitoring settings. I had to manually add the software file to the list.

**Comment by rnayabed** [comment]
Thank you for reporting. Did you try both the portable and installer?

**Comment by pratyushmanocha** [comment]
Yep. The portable one updates the log with the errors (the file I uploaded in the previous comment), whereas the installer version doesn't update the log with the actions and errors post the initial update checking stage; I tried changing the key bindings and the rgb pattern, it failed and gave the same permission issues error, however nothing got updated in the log file (check attached file). 
[rangoli.log](https://github.com/rnayabed/rangoli/files/11181316/rangoli.log)

**Comment by pratyushmanocha** [comment]
Hey Debayan, any updates on this issue?

**Comment by rnayabed** [comment]
Sorry, I have to check

**Comment by rnayabed** [comment]
Need to setup a moterenery vm then will try it. Was busy lately

**Comment by pratyushmanocha** [comment]
Sure, no worries. Thanks for the hard work!

**Comment by k8seren** [comment]
Hi just wanted to note the same issue, MacOS 13.3.1(a) - the app says insufficient permissions, but I manually added Rangoli.app to the Input Monitoring settings and set it to allow. 
Additionally the app does not quit normally. It hangs on quit and must be force quit to actually close. Maybe this is related?

I rebooted and no change. Happy to provide any details to troubleshoot - I'd love to see this working on MacOS! 🥇 

Using MBP M1, 13.3.1(a), and an RK68Pro (which shows up in the application immediately, awesome!)

**Comment by rnayabed** [comment]
Can you provide the rangoli.log file saved in your user home directory?
thanks

बुध, 17 मई 2023 को 9:08 pm बजे को k8seren ***@***.***> ने
लिखा:

**Comment by k8seren** [comment]
Sure thing, here it is.
[rangoli.log](https://github.com/rnayabed/rangoli/files/11510828/rangoli.log)

**Comment by smweil** [comment]
Bumping this with the same issue...

**Comment by thegoldknight** [comment]
Bumping also with the same issue

**Comment by Artiaure** [comment]
Hi ! did anyone manage to find a fix to this problem ?

**Comment by BilalN777** [comment]
Hey @rnayabed any update?

**Comment by BilalN777** [comment]
I may have found a way to reproduce the issue


I've found that if karabiner is running then I get the permission issues

if I quit karabiner is works just find without having to even relaunch rangoli

as soon I launch karabiner again and attempt to save a color setting on rangoli it gives me permission issues

**Comment by pratyushmanocha** [comment]
Hi, @rnayabed, 

Did you get the time to look at this issue and possibly fix it?

Thanks!
